### PR TITLE
#738 - Fix path handling for NpmProxySlice

### LINKED ISSUE #738: Fix path handling for NpmProxy
If in `SliceFromConfig`  use `""` instead of `cfg.path()` then `meta.json` will be found by proxy and loaded but `.tgz` will not be found (output below). If use `cfg.path()` then neither of the two files will be found:
```
            case "npm-proxy":
                slice = trimIfNotStandalone(
                    settings, standalone,
                    new NpmProxySlice(
//                        cfg.path(),
                        "",
                        new NpmProxy(
                            new NpmProxyConfig(cfg.settings().orElseThrow()),
                            Vertx.vertx(),
                            cfg.storage()
                        )
                    )
                );
                break;
```

```
[INFO] main com.artipie.VertxMain: Artipie was started on port 65447
[INFO] FileStorage-3 com.artipie.asto.fs.FileStorage: Found 4 objects by the prefix "" in C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\origin\repos by C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\origin\repos: [_credentials.yml, npm-origin.yaml, npm-origin/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz, npm-origin/@hello/simple-npm-project/meta.json]
[DEBUG] FileStorage-5 com.artipie.RepoConfig: parsed yaml config:
repo:
  type: npm
  storage:
    type: fs
    path: "C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\origin\repos"
  url: "http://localhost:65447/npm-origin"
[INFO] main com.artipie.VertxMain: Artipie was started on port 65482
[INFO] FileStorage-6 com.artipie.asto.fs.FileStorage: Found 2 objects by the prefix "" in C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\proxy\repos by C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\proxy\repos: [_credentials.yml, my-npm.yaml]
[DEBUG] FileStorage-7 com.artipie.RepoConfig: parsed yaml config:
repo:
  type: "npm-proxy"
  storage:
    type: fs
    path: "C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\proxy\repos"
  url: "http://host.testcontainers.internal:65482/my-npm/"
  path: "my-npm"
  settings:
    remote:
      url: "http://localhost:65447/npm-origin"
[INFO] vert.x-eventloop-thread-1 com.artipie.http.MainSlice: >> GET /npm-origin/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz HTTP_1_1

User-Agent: Java/11.0.8
Host: localhost:65447
Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2
Connection: keep-alive
[DEBUG] FileStorage-5 com.artipie.RepoConfig: parsed yaml config:
repo:
  type: npm
  storage:
    type: fs
    path: "C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\origin\repos"
  url: "http://localhost:65447/npm-origin"
[INFO] FileStorage-1 com.artipie.http.auth.Permissions$$Lambda$660/0x00000008004dac40: Operation 'r' allowed for '*'
[INFO] FileStorage-2 com.artipie.asto.fs.FileStorage: Exists 'npm-origin/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz': true
[INFO] FileStorage-8 com.artipie.asto.fs.FileStorage: Value 'npm-origin/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz': Optional[307]
[INFO] FileStorage-8 com.artipie.http.MainSlice: << OK
[DEBUG] FileStorage-4 com.artipie.MeasuredSlice: GET /npm-origin/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz HTTP_1_1
: response=1ms accept=10ms send=173ms total=177ms
[DEBUG] main com.artipie.test.TestContainer: Command:
npm install @hello/simple-npm-project --registry http://host.testcontainers.internal:65482/my-npm/
[INFO] vert.x-eventloop-thread-1 com.artipie.http.MainSlice: >> GET /my-npm/@hello%2fsimple-npm-project HTTP_1_1

connection: keep-alive
user-agent: npm/6.14.8 node/v14.14.0 linux x64
npm-in-ci: false
npm-scope: 
npm-session: b44f24862426afb7
referer: install [REDACTED]
pacote-req-type: packument
pacote-pkg-id: registry:@hello/simple-npm-project
accept: application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*
accept-encoding: gzip,deflate
Host: host.testcontainers.internal:65482
[DEBUG] FileStorage-7 com.artipie.RepoConfig: parsed yaml config:
repo:
  type: "npm-proxy"
  storage:
    type: fs
    path: "C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\proxy\repos"
  url: "http://host.testcontainers.internal:65482/my-npm/"
  path: "my-npm"
  settings:
    remote:
      url: "http://localhost:65447/npm-origin"
[DEBUG] FileStorage-2 com.artipie.npm.proxy.http.DownloadPackageSlice: >> GET /@hello/simple-npm-project HTTP_1_1

connection: keep-alive
user-agent: npm/6.14.8 node/v14.14.0 linux x64
npm-in-ci: false
npm-scope: 
npm-session: b44f24862426afb7
referer: install [REDACTED]
pacote-req-type: packument
pacote-pkg-id: registry:@hello/simple-npm-project
accept: application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*
accept-encoding: gzip,deflate
Host: host.testcontainers.internal:65482
X-FullPath: /my-npm/@hello/simple-npm-project
[DEBUG] FileStorage-2 com.artipie.npm.proxy.http.PackagePath: Determined path is: @hello/simple-npm-project
[INFO] FileStorage-3 com.artipie.asto.fs.FileStorage: Exists 'my-npm/@hello/simple-npm-project/meta.json': false
[INFO] vert.x-eventloop-thread-1 com.artipie.http.MainSlice: >> GET /npm-origin/@hello/simple-npm-project HTTP_1_1

user-agent: Artipie
host: localhost:65447
[DEBUG] FileStorage-1 com.artipie.RepoConfig: parsed yaml config:
repo:
  type: npm
  storage:
    type: fs
    path: "C:\Users\AWX971~1\AppData\Local\Temp\junit13188099467082796481\origin\repos"
  url: "http://localhost:65447/npm-origin"
[INFO] FileStorage-1 com.artipie.http.auth.Permissions$$Lambda$660/0x00000008004dac40: Operation 'r' allowed for '*'
[INFO] FileStorage-5 com.artipie.asto.fs.FileStorage: Exists 'npm-origin/@hello/simple-npm-project/meta.json': true
[INFO] FileStorage-6 com.artipie.asto.fs.FileStorage: Value 'npm-origin/@hello/simple-npm-project/meta.json': Optional[879]
[INFO] FileStorage-6 com.artipie.http.MainSlice: << OK
Content-Type: application/json
[DEBUG] FileStorage-7 com.artipie.MeasuredSlice: GET /npm-origin/@hello/simple-npm-project HTTP_1_1
: response=0ms accept=34ms send=46ms total=46ms
[INFO] FileStorage-2 com.artipie.asto.fs.FileStorage: Save 'my-npm/@hello/simple-npm-project/meta.json': Optional[879]
[INFO] FileStorage-7 com.artipie.asto.fs.FileStorage: Save 'my-npm/@hello/simple-npm-project/meta.meta': Optional[75]
[DEBUG] FileStorage-7 com.artipie.npm.proxy.http.DownloadPackageSlice: << OK
Content-Type: application/json
Last-Modified: null
Content-Length: 920
[INFO] FileStorage-7 com.artipie.http.MainSlice: << OK
Content-Type: application/json
Last-Modified: null
Content-Length: 920
[DEBUG] FileStorage-7 com.artipie.MeasuredSlice: GET /my-npm/@hello%2fsimple-npm-project HTTP_1_1
: response=0ms accept=2ms send=476ms total=477ms
[INFO] vert.x-eventloop-thread-1 com.artipie.http.MainSlice: >> GET /@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz HTTP_1_1

connection: keep-alive
user-agent: npm/6.14.8 node/v14.14.0 linux x64
npm-in-ci: false
npm-scope: 
npm-session: b44f24862426afb7
referer: install [REDACTED]
pacote-req-type: tarball
pacote-pkg-id: registry:@hello/simple-npm-project@http://host.testcontainers.internal:65482/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz
accept: */*
Host: host.testcontainers.internal:65482
[INFO] FileStorage-3 com.artipie.http.MainSlice: << NOT_FOUND
Content-Length: 29
[DEBUG] FileStorage-3 com.artipie.MeasuredSlice: GET /@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz HTTP_1_1
: response=0ms accept=0ms send=2ms total=2ms


java.lang.AssertionError: Package was installed
Expected: a string containing "+ @hello/simple-npm-project@1.0.1", "added 1 package" in order
     but: was "npm ERR! code E404npm ERR! 404 Not Found - GET http://host.testcontainers.internal:65482/@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgznpm ERR! 404 npm ERR! 404  '@hello/simple-npm-project@latest' is not in the npm registry.npm ERR! 404 You should bug the author to publish it (or use the name yourself!)npm ERR! 404 npm ERR! 404 Note that you can also install from anpm ERR! 404 tarball, folder, http url, or git url.npm ERR! A complete log of this run can be found in:npm ERR!     /root/.npm/_logs/2020-11-05T06_40_13_746Z-debug.log"
```
So, path handling does not work properly. It should be fixed. After that `NpmProxyITCase` should be disabled.

### PR DESCRIPTION
**Pull request by genryxy**
Closes #738 
The problem was that the `TrimPathSlice` cut off repository name from the request therefore patterns that were created before in `NpmProxySlice` became invalid. Refusing to use `trimIfNotStandalone` in `SliceFromConfig` for `npm-proxy` helped to solve this problem.

### CODE REVIEW

File: src/main/java/com/artipie/SliceFromConfig.java [hunk 1]
```diff
@@ -250,15 +250,12 @@ static Slice build(
...
-                        new NpmProxy(
-                            new NpmProxyConfig(cfg.settings().orElseThrow()),
-                            Vertx.vertx(),
-                            cfg.storage()
-                        )
+                slice = new NpmProxySlice(
+                    cfg.path(),
+                    new NpmProxy(
+                        new NpmProxyConfig(cfg.settings().orElseThrow()),
+                        Vertx.vertx(),
```
Comment (olegmoz) [inline src/main/java/com/artipie/SliceFromConfig.java:257]: @genryxy it is not something you added, but there is a big problem in this code. `Vertx.vertx()` call creates a Vert.x instance, a very heavy object. It should be created only once for whole application on start an reused everywhere. Could you please add a puzzle for solving this issue?

**Review by olegmoz — CHANGES_REQUESTED** [review]
@genryxy thanks, please see one comment

**Comment by genryxy** [comment]
@olegmoz thanks, add todo

**Review by olegmoz — APPROVED** [review]
@genryxy thanks, merging

**Pull request merged**